docker/gcp_pudl_etl.sh

```diff
@@ -31,11 +31,17 @@ function run_pudl_etl() {
     && pudl_etl \
         --loglevel DEBUG \
         --gcs-cache-path gs://internal-zenodo-cache.catalyst.coop \
-        $PUDL_SETTINGS_YML \
-    && pytest \
+        $PUDL_SETTINGS_YML
+    # Run unit+integration tests and data validations in parallel in the background
+    # They each take ~1 hour on their own and have no interdependencies.
+    pytest \
+        --gcs-cache-path=gs://internal-zenodo-cache.catalyst.coop \
+        --etl-settings=$PUDL_SETTINGS_YML \
+        --live-dbs test/unit test/integration &
+    pytest \
```

Comment:
Is there a specific reason for controlling the boundaries for parallel/serial execution in this manner over something like using `pytest-xdist` and parallelizing everything? https://www.tutorialspoint.com/pytest/pytest_run_tests_in_parallel.htm

Reply:
I had thought there was *some* barrier to using `pytest-xdist` before (hence suggesting `tox run-parallel`), but @zaneselvans might know better.

When I experimented with splitting up tests across multiple workers using `pytest-shard`, I found that each test process still had to instantiate some very expensive test fixtures, so that didn't end up improving our times that much...

Reply:
A big majority of the time that the tests spend is setting up fixtures, and IIRC when we messed around with `pytest-shard` and `pytest-xdist` it resulted in many different copies of the same time-intensive fixtures getting set up, which didn't save any time.

The reason why this might work well for the `validate` and `full_integration` tests is because they mostly just read out of a pre-populated database (b/c `--live-dbs`) so I think any of these parallelization schemes would probably be okay.  I don't think we're going to get a ton of benefit from throwing more than 2 CPUs at this problem right now because both of these tests still depend on `PudlTabl` fixtures which are resource intensive. But as more of what was being done in `PudlTabl` is written directly into the DB, more CPUs will be helpful.

We could use `tox -p` which would split up the tests into 2 groups (integration + validate) well, but right now I don't think anything in the nightly builds are being run using Tox, because the Docker container already has the required Python environment in it. There are [a couple of different ways](https://pytest-xdist.readthedocs.io/en/stable/distribution.html) to distribute tests to workers with `pytest-xdist` by module, file, or marked test group.